src/xcode/ENA/ENA/Source/Services/HealthCertificateValidationService/Model/HealthCertificateValidationError.swift

```diff
@@ -22,4 +22,35 @@ enum HealthCertificateValidationError: Error {
 	case NO_NETWORK
 	case VALUE_SET_SERVER_ERROR
 
+	var errorDescription: String? {
+		switch self {
+		case .TECHNICAL_VALIDATION_FAILED:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (TECHNICAL_VALIDATION_FAILED)"
+		case .ACCEPTANCE_RULE_CLIENT_ERROR:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (ACCEPTANCE_RULE_CLIENT_ERROR)"
+		case .ACCEPTANCE_RULE_JSON_ARCHIVE_FILE_MISSING:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (ACCEPTANCE_RULE_JSON_ARCHIVE_FILE_MISSING)"
+		case .ACCEPTANCE_RULE_JSON_ARCHIVE_SIGNATURE_INVALID:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (ACCEPTANCE_RULE_JSON_ARCHIVE_SIGNATURE_INVALID)"
+		case .ACCEPTANCE_RULE_JSON_EXTRACTION_FAILED:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (ACCEPTANCE_RULE_JSON_EXTRACTION_FAILED)"
+		case .ACCEPTANCE_RULE_SERVER_ERROR:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (ACCEPTANCE_RULE_SERVER_ERROR)"
+		case .INVALIDATION_RULE_CLIENT_ERROR:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (INVALIDATION_RULE_CLIENT_ERROR)"
+		case .INVALIDATION_RULE_JSON_ARCHIVE_FILE_MISSING:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (INVALIDATION_RULE_JSON_ARCHIVE_FILE_MISSING)"
+		case .INVALIDATION_RULE_JSON_ARCHIVE_SIGNATURE_INVALID:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (INVALIDATION_RULE_JSON_ARCHIVE_SIGNATURE_INVALID)"
+		case .INVALIDATION_RULE_JSON_EXTRACTION_FAILED:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (INVALIDATION_RULE_JSON_EXTRACTION_FAILED)"
+		case .INVALIDATION_RULE_SERVER_ERROR:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (INVALIDATION_RULE_SERVER_ERROR)"
+		case .NO_NETWORK:
+			return "\(AppStrings.HealthCertificate.ValidationError.noNetwork) (NO_NETWORK)"
+		case .VALUE_SET_SERVER_ERROR:
+			return "\(AppStrings.HealthCertificate.ValidationError.tryAgain) (VALUE_SET_SERVER_ERROR)"
+		}
```

Comment:
Note: While writing on the rule validation, it turned out that there will be two more error cases, for each rule type one. I will add a jira subtask as a reminder that we don't forget to add them here.

Reply:
I think we will run into a compile-time error anyway if we are missing cases, but it's good to have the task just to make sure 👍